WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -1659,12 +1661,8 @@ private void updateAndSavePostAsyncOnEditorExit(@NonNull final OnPostUpdatedFrom
         // Store this before calling updateAndSavePostAsync because its value can change before the callback returns
         boolean isAutosavePending = mViewModel.isAutosavePending();
 
-        mEditorFragment.showSavingProgressDialogIfNeeded();
+        mViewModel.showSaveProgressDialog();
         updateAndSavePostAsync((result) -> {
-            if (mEditorFragment != null) {
-                mEditorFragment.hideSavingProgressDialog();
```

Comment:
Because of the `LiveData` and `GutenbergEditorFragment::onDestroy` changes, the Activity doesn't need to worry about hiding the dialog anymore.